mysite/ct/views.py

```diff
@@ -1033,6 +1033,9 @@ def edit_lesson(request, course_id, unit_id, ul_id):
     else: # let instructor edit this lesson
         titleform = formClass(instance=ul.lesson, initial=dict(changeLog=''))
         if request.method == 'POST':
+            if 'unit_to_move' in request.POST:
+                ul.unit = Unit.objects.get(id=request.POST['unit_to_move'])
+                ul.save()
```

Comment:
Return redirect to new url - this need to be edit lesson page with new `unit_id`.